Src/RE2.Managed/Regex2.cs

```diff
@@ -187,111 +187,71 @@ public static void ClearRegexes()
             }
         }
 
-        /// <summary>
-        /// Searches the text for the specified pattern.
-        ///
-        /// For simplicity, the implementation uses 32-bit signed integers throughout. There is no size-related error checking.
-        /// Hence, if some count or size exceeds that (e.g. number of named groups, length of text), there will be problems.
-        /// </summary>
-        ///
-        /// <param name="pattern">Pattern to search for in RE2 syntax.</param>
-        /// <param name="text">Text to search.</param>
-        /// <param name="groupName2Index">Map of group name to index in <paramref name="submatchStrings"/>.</param>
-        /// <param name="index2GroupName">Map of index in <paramref name="submatchStrings"/> to group name.</param>
-        /// <param name="submatchStrings">List of texts for each matching group.</param>
-        ///
-        /// <returns>Boolean indicating if the pattern matches the text.</returns>
-        ///
-        /// <example>
-        /// <code>
-        /// Input pattern @"(?P<g1>a)(b)(?P<g2>c)"
-        /// Input text = @"abc"
-        ///
-        /// groupName2Index = { "g1": 0, "g2": 2 }
-        /// index2GroupName = { 0: "g1", 2: "g2" }
-        /// index2GroupName = [ "abc", "a", "b", "c" ]
-        /// </code>
-        /// </example>
-        public static unsafe bool Matches(
-            string pattern,
-            string text,
-            out Dictionary<string, int> groupName2Index,
-            out Dictionary<int, string> index2GroupName,
-            out List<string> submatchStrings)
+        public static unsafe void MatchesNamedGroups2(string pattern, string text, out List<Dictionary<string, string>> matches)
```

Comment:
>MatchesNamedGroups2 [](http://example.com/codeflow?start=34&length=19)

Add documentation

Reply:
I think we could use Matches as name, since the overload would work

Reply:
Fixed.